python/BioSimSpace/Sandpit/Exscientia/_SireWrappers/_molecule.py

```diff
@@ -1706,29 +1713,46 @@ def _toRegularMolecule(
                 # Search for any dummy atoms.
                 try:
                     search = mol.atoms("element Xx")
-                except:
+                except KeyError:
                     search = []
 
                 # Replace the ambertype.
                 for dummy in search:
                     index = dummy.index()
+                    amber_type_value = amber_types[index.value()]
+                    element_value = elements[index.value()]
+
+                    # We have a dummy in both endstates so we try to infer the element.
+                    # This is not general so it is only suitable for some common cases.
+                    if element_value.symbol() == "Xx":
+                        name = property_map.get("name", "name")
+                        element_symbol = dummy.property(name)[0].upper()
+                        element_value = _SireMol.Element(element_symbol)
+
                     mol = (
                         mol.atom(index)
-                        .setProperty(amber_type, amber_types[index.value()])
-                        .molecule()
-                    )
-                    mol = (
-                        mol.atom(index)
-                        .setProperty(element, elements[index.value()])
+                        .setProperty(amber_type, amber_type_value)
                         .molecule()
                     )
+                    mol = mol.atom(index).setProperty(element, element_value).molecule()
 
                 # Delete redundant properties.
                 mol = mol.removeProperty("ambertype0").molecule()
                 mol = mol.removeProperty("ambertype1").molecule()
                 mol = mol.removeProperty("element0").molecule()
                 mol = mol.removeProperty("element1").molecule()
 
+        if generate_intrascale:
+            # First we regenerate the connectivity based on the bonds.
+            conn = _SireMol.Connectivity(mol.info()).edit()
+            for bond in mol.property("bond").potentials():
+                conn.connect(bond.atom0(), bond.atom1())
+            mol.setProperty("connectivity", conn.commit())
+
+            # Now we have the correct connectivity, we can regenerate the exclusions.
+            gro_sys = _SireIO.GroTop(_System(mol)._sire_object).toSystem()
```

Comment:
How does Gromacs come into play here?

Reply:
This is just a hack to regenerate the exclusion list, I think it has nothing to do with Gromacs per se. Lester uses it in a few places so I have just copied the approach.